### ISSUE TITLE
panic when a wallet is not connected

### ISSUE THREAD

**Issue by jolan** [decred/dcrstakepool#166]
@mverrilli reported a panic:

goroutine 42 [running]:
panic(0xa4c740, 0xc4205e8b40)
/usr/local/go/src/runtime/panic.go:500 +0x1a1
github.com/decred/dcrstakepool/controllers.(*walletSvrManager).walletRPCHandler(0xc420452480)
/home/dcr/go/src/github.com/decred/dcrstakepool/controllers/dcrclient.go:198 +0x41c
created by github.com/decred/dcrstakepool/controllers.(*walletSvrManager).Start
home/dcr/go/src/github.com/decred/dcrstakepool/controllers/dcrclient.go:1014 +0x12e

The code in question is:

```go
			case connectedMsg:
				resp := w.executeInSequence(connectedFn, msg)
				respTyped := resp.(*connectedResponse)
				msg.reply <- *respTyped
```

He had turned on txindex=1 in dcrd so possibly dcrwallet was erroring out since it wasn't connected rather than sending back a proper walletinfo response indicating it wasn't connected.

**Comment by mverrilli** [comment]
In case it helps, this is what the wallet daemon looked like during the txindex rebuild: 

```
14:09:33 2017-05-04 [INF] RPCS: Disconnected websocket client xxx.xxx.xxx.xxx:xxxxx
14:09:39 2017-05-04 [INF] RPCS: New websocket client xxx.xxx.xxx.xxx:xxxxx
14:09:39 2017-05-04 [INF] RPCS: RPC method walletinfo invoked by client xxx.xxx.xxx.xxx:xxxxx
14:09:40 2017-05-04 [INF] RPCS: RPC method walletinfo invoked by client xxx.xxx.xxx.xxx:xxxxx
14:09:40 2017-05-04 [WRN] RPCS: Websocket receive failed from client xxx.xxx.xxx.xxx:xxxxx: websocket: close 1006 unexpected EOF
14:09:40 2017-05-04 [INF] RPCS: Disconnected websocket client xxx.xxx.xxx.xxx:xxxxx
14:09:41 2017-05-04 [ERR] CHNS: Connection attempt failed: dial tcp yyy.yyy.yyy.yyy:9109: getsockopt: connection refused
14:10:01 2017-05-04 [ERR] CHNS: Connection attempt failed: dial tcp yyy.yyy.yyy.yyy:9109: getsockopt: connection refused
```

**Comment by jholdstock** [comment]
@dajohi  I believe this can be closed thanks to #494

**Comment by chappjc** [comment]
I concur.
